app/src/main/java/org/oppia/android/app/story/StoryFragmentPresenter.kt

```diff
@@ -142,6 +150,40 @@ class StoryFragmentPresenter @Inject constructor(
             ).parseOppiaHtml(
               storyItemViewModel.summary, binding.chapterSummary
             )
+          if (storyItemViewModel.chapterSummary.chapterPlayState
+            == ChapterPlayState.NOT_PLAYABLE_MISSING_PREREQUISITES
+          ) {
+            val missingPrerequisiteSummary = fragment.getString(
+              R.string.chapter_prerequisite_title_label,
+              storyItemViewModel.index.toString(),
+              storyItemViewModel.missingPrerequisiteChapter.name
+            )
+            val chapterLockedSpannable = SpannableString(missingPrerequisiteSummary)
+            val clickableSpan = object : ClickableSpan() {
+              override fun onClick(widget: View) {
+                smoothScrollToPosition(storyItemViewModel.index - 1)
+              }
+
+              override fun updateDrawState(ds: TextPaint) {
+                super.updateDrawState(ds)
+                ds.isUnderlineText = false
+              }
+            }
+            chapterLockedSpannable.setSpan(
+              clickableSpan,
+              9,
+              chapterLockedSpannable.length - 24,
```

Comment:
`/* end= */chapterLockedSpannable.length - 24,`

Also, please mention why you are subtracting exactly 24 only?

Reply:
Similar to the approach mentioned in the https://github.com/oppia/oppia-android/pull/2948#discussion_r600244059, this time we have to count from the end. And on doing so, we will find that there are **24** characters between the second placeholder `%s` and the end. Thus, we are subtracting **24** from the length of the final string. Please share your thoughts on this as well. Thanks.

Also, I have added the inline comments. PTAL.